### PR TITLE
Only build image if changes in the proper folders

### PR DESCRIPTION
**Pull request by LanderU**
Signed-off-by: LanderU <lander.usategui@gmail.com>

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,12 +1,31 @@
...
+  # For pull requests it's not necessary to checkout the code
+  - uses: dorny/paths-filter@v2
+    id: filter
+    with:
+      filters: |
+        check-changes:
+          - 'builder/**''
+          - 'scripts/**'
+          - 'focal-rt-ros2/**'
+          - 'Makefile'
```
Comment (shuhaowu) [inline .github/workflows/build.yml:24]: Probably also when the .github folder is changed in case these build.yml is changed?

**Comment by shuhaowu** [comment]
Also the build is broken right now because Canonical removed `https://cdimage.ubuntu.com/releases/20.04.3/release/ubuntu-20.04.3-preinstalled-server-arm64+raspi.img.xz`. They've released the 20.04.4 point release. 

I'm fixing it in my refactor with Python right now. So this PR can wait until that is done (hopefully soon).

File: .github/workflows/build.yml (hunk 1 above)
Comment (LanderU) [reply .github/workflows/build.yml:24]: Yep, good catch, will add this as well!

**Comment by LanderU** [comment]
@shuhaowu, do they have mirrors for old images? If not, we can maybe create a nightly job to check if the image is not available.

**Comment by carlossvg** [comment]
@LanderU @shuhaowu If this is a recurrent problem we can consider storing the image internally as a release.

**Comment by LanderU** [comment]
@carlossvg, yes, we should do so.

File: .github/workflows/build.yml (hunk 1 above)
Comment (LanderU) [reply .github/workflows/build.yml:24]: I need to refactor this PR

**Comment by carlossvg** [comment]
@shuhaowu @LanderU I uploaded the Ubuntu image in the release section (see https://github.com/ros-realtime/ros-realtime-rpi4-image/releases/tag/untagged-221fb272d1e86ed09c57). However, I have the feeling it doesn't make much sense to have it as a separate release. Maybe we could do a release an upload all the images there, including the Ubuntu base image.

**Comment by carlossvg** [comment]
@LanderU What is the status of this PR? Is it ready to merge?

**Comment by LanderU** [comment]
@carlossvg, not yet, I need to convert this to use the new python style.

**Comment by LanderU** [comment]
@carlossvg @shuhaowu, would you check again? Thanks in advance!

**Review by shuhaowu — APPROVED** [review]

**Pull request merged**